frontend/src/metabase/parameters/components/ParameterSettings/ParameterSettings.tsx

```diff
@@ -46,11 +46,35 @@ const ParameterSettings = ({
   onChangeSourceConfig,
   onRemoveParameter,
 }: ParameterSettingsProps): JSX.Element => {
+  const [internalValue, setInternalValue] = useState(parameter.name);
+
+  useLayoutEffect(() => {
+    setInternalValue(parameter.name);
```

Comment:
[It was not always impossible to have empty parameter name](https://github.com/metabase/metabase/pull/10323), so there may be dashboards out there in the wild with empty parameter names.
If someone with such dashboard tried to upgrade from 0.32, then they would not be able to change the label.

```suggestion
    if (parameter.name) {
      setInternalValue(parameter.name);
    }
```

Reply:
It's now impossible to have an empty parameter name. I just fixed it with #34311.

Reply:
Oh, cool. I though we're not doing it because of [this comment](https://github.com/metabase/metabase/issues/27653#issuecomment-1422033860).